### ISSUE TITLE
Prometheus: Incorrect variable results when label value contains quote

### ISSUE THREAD

**Issue by jmmorel** [grafana/grafana#16275]
**What happened**:
Variables values depending on other variable value are not found when the latter contains quote

**What you expected to happen**:
Expects grafana to find values

**How to reproduce it (as minimally and precisely as possible)**:
On a dashboard, let's define two variables from a prometheus datasource:
group: label_values(apache_up, group)
env: label_values(apache_up{group="$group"}, env)

When the selected group value contains a quote, the corresponding env values are not populated.

**Anything else we need to know?**:
Let's say there is the following group value "winner's group".
The resulting query is :
/api/datasources/proxy/29/api/v1/series?match[]=apache_up%7Bgroup%3D%22winners%5C%5C%27group%22%7D&start=1553770648&end=1553774248
which doesn't work

In the variables definition screen, when setting the hardcoded value label_values(apache_up{group="winners'group"}, env), the resulting query is:
/api/datasources/proxy/29/api/v1/series?match[]=apache_up%7Bgroup%3D%22winners%27group%22%7D&start=1553778535&end=1553782135
This one works

It seems Grafana unnecessarily escapes the quote with \\\\

**Environment**:
- Grafana version: 5.4.2 & 6.0.0
- Data source type & version: prometheus 2.7.1
- OS Grafana is installed on: docker 17.03 on RHEL 7.3 
- User OS & Browser:Win7 Chrome/Firefox/...
- Grafana plugins:None
- Others:

**Comment by dalbani** [comment]
I've run into this issue today, when trying to visualize the following metric with a label on `pool`:
```jvm_memory_pool_bytes_used{pool="CodeHeap 'non-nmethods'",} 1342080.0```
What matters here are the single quotes in the name.

With logging of the proxy feature activated, I could see the following request:
```
/api/datasources/proxy/1/api/v1/query_range?query=jvm_memory_pool_bytes_used%7Binstance%3D%22d6.acc.geodan.nl%3A8984%22%2C%20pool%3D%22CodeHeap%20%5C%5C%27non-nmethods%5C%5C%27%22%7D&start=1562763540&end=1562767170&step=30"
```
Or, more legible, as URL decoded:
```
/api/datasources/proxy/1/api/v1/query_range?query=jvm_memory_pool_bytes_used{instance="d6.acc.geodan.nl:8984", pool="CodeHeap \\'non-nmethods\\'"}&start=1562763540&end=1562767170&step=30
```
A double backslash is used to escape the single quote.
That's due to a method defined in https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/prometheus/datasource.ts#L727.
It was introduced in commit https://github.com/grafana/grafana/commit/1f1719c49822d2a43f225143bb6d389408ad7b43 to fix bugs #10555 and #6888, I see now.
I followed @torkelo's advice in the second bug and used the `raw` suffix successfully, as in:
```
          "expr": "jvm_memory_pool_bytes_used{instance=\"$instance\", pool=\"${jvm_memory_pool:raw}\"}",
```
I hope that this information can be useful to you as well, @jmmorel.

**Comment by ivanahuckova** [comment]
Closing as this. is duplicate of  #6888. As mentioned in #6888 and in comment above, it is recommended to use the `=~`operator in your prometheus query. Or change the template variable options and remove the Multi & All options (they cause the values to be regex formated as that is the only way multi values can be filtered). Another option would be to use regex operator in query or ${varnade:raw} - more on this here: https://grafana.com/docs/grafana/latest/variables/advanced-variable-format-options/#raw.
